ui/src/image-uploading/uploader.js

```diff
@@ -0,0 +1,46 @@
+import React, { Component } from 'react'
+import axios from 'axios';
+
+// const client_id = "f4ca50bc021558a" //"ff3a3f659d1a390"
+
+class App extends Component {
+   state = {
+      selectedFile: null,
+      url: null
+   }
+
+   selectFileHelper = event => {
+      this.setState({selectedFile: event.target.files[0]});
+      this.setState({url: URL.createObjectURL(event.target.files[0])});
+   }
+
```

Comment:
This might help. From my old project.  
  const handleSubmit = variables => {
  	  const request = new XMLHttpRequest();
	  const data = new FormData();
	  const image = document.getElementsByClassName('input-image')[0].files[0];

	  let clientID = "e709c64107023d2";
	  let imageURL;
	  let result;

	  data.append('image', image);

	  request.open('POST', 'https://api.imgur.com/3/image/');
	  request.setRequestHeader('Authorization', 'Client-ID ' + clientID);
	  request.onreadystatechange = function () {
	     if(request.status === 200 && request.readyState === 4) {
		    result = JSON.parse(request.responseText);
		    imageURL = 'https://i.imgur.com/' + result.data.id + '.png';
		    console.log(imageURL);
		    const adFields = { name, primarytext, subtext, redirectURL, imageURL };
		    postToBackend(adFields);
	     }
	  }
	  request.send(data);

      setName("");
      setPrimaryText("");
      setSubtext("");
      setRedirectURL("");
      setSelectedImage("");
   };

Reply:
That code was something similar to what I had working, however, I was getting 400 and 429 errors, and wasn't able to figure out the axios and imgur api